File: Heuristic Search Planner (sgware)/src/edu/uno/ai/planning/hsp/HeuristicSearchPlanner.java
package edu.uno.ai.planning.hsp;

import edu.uno.ai.planning.h.CompleteSearch;
import edu.uno.ai.planning.h.StateHeuristic;
import edu.uno.ai.planning.h.HeuristicPlanner;
import edu.uno.ai.planning.h.HeuristicSearch;
import edu.uno.ai.planning.h.HillClimbingSearch;
import edu.uno.ai.planning.ss.StateSpaceProblem;

public class HeuristicSearchPlanner extends HeuristicPlanner {
	
	public HeuristicSearchPlanner() {
		super("SHSP");
	}

	@Override
	protected HeuristicSearch makeStateSpaceSearch(StateSpaceProblem problem) {
		StateHeuristic heuristic = new AdditiveHeuristic(problem);
		return new CompleteSearch(problem, heuristic, HeuristicSearch.A_STAR);
	}
}

File: UnitySokoban/Assets/Scripts/Planning/HeuristicSearchPlannerSGW/HeuristicSearchPlanner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Planning;
using StateSpaceSearchProject;

namespace HeuristicSearchPlannerSGW
{
    public class HeuristicSearchPlanner : HeuristicPlanner
    {
        public HeuristicSearchPlanner()
            :base("SHSP")
        {
        }

        public override HeuristicSearch makeSearch(Problem problem)
        {
            StateHeuristic heuristic = new AdditiveHeuristic((StateSpaceProblem)problem);
            return new CompleteSearch((StateSpaceProblem)problem, heuristic, HeuristicSearch.A_STAR);
        }
    }
}
